DylibBinder/DBParameter.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using SwiftReflector;
+
+namespace DylibBinder {
+	internal class DBParameter {
+		public string Type { get; set; }
+		public string PublicName { get; }
+		public string PrivateName { get; }
+		public bool IsVariadic { get; }
+		public int Index { get; }
+		public bool HasInstance { get; }
+		public bool IsConstructor { get; }
+		public bool IsEmptyParameter { get; }
+
+		public DBParameter (string type, string publicName, string privateName, int index, ParameterOptions parameterOptions)
+		{
+			Type = type;
+			PublicName = publicName;
+			PrivateName = privateName;
+			Index = index;
+			IsVariadic = parameterOptions.HasFlag (ParameterOptions.IsVariadic);
+			HasInstance = parameterOptions.HasFlag (ParameterOptions.HasInstance);
+			IsConstructor = parameterOptions.HasFlag (ParameterOptions.IsConstructor);
+			IsEmptyParameter = parameterOptions.HasFlag (ParameterOptions.IsEmptyParameter);
+		}
+
+		public static DBParameter CreateInstanceParameter () => new DBParameter ("", "", "self", 0, ParameterOptions.HasInstance);
+		public static DBParameter CreateInstanceParameter (int index) => new DBParameter ("", "", "self", index, ParameterOptions.HasInstance);
+		public static DBParameter CreateConstructorParameter (int index) => new DBParameter (".Type", "", "self", index, ParameterOptions.IsConstructor);
+		public static DBParameter CreateGetterParameter () => new DBParameter (null, null, null, 0, ParameterOptions.IsEmptyParameter);
+		public static DBParameter CreateSetterParameter () => new DBParameter ("", "", "self", 0, ParameterOptions.HasInstance);
+	}
+
+
+	internal class DBParameterList : IAssociatedTypes {
+		public int Index { get; }
+		public List<DBParameter> ParameterCollection { get; } = new List<DBParameter> ();
+		public DBAssociatedTypes AssociatedTypes { get; } = new DBAssociatedTypes ();
+
+		public DBParameterList (SwiftBaseFunctionType signature, bool hasInstance, bool isConstructor, int index)
+		{
+			Index = index;
+			int parameterIndex = 0;
+			AssociatedTypes.AssociatedTypeCollection.UnionWith (signature.GetAssociatedTypes ());
+
+			if (hasInstance || isConstructor) {
+				ParameterCollection.Add (hasInstance ? DBParameter.CreateInstanceParameter (parameterIndex) : DBParameter.CreateConstructorParameter (parameterIndex));
+				return;
+			}
+
+			foreach (var parameter in signature.EachParameter) {
+				var type = SwiftTypeToString.MapSwiftTypeToString (parameter);
+				var publicName = "_";
+				var privateName = "privateName";
+
+				if (parameter.Name != null)
+					publicName = privateName = parameter.Name.Name;
```

Comment:
Same can be achieved with ??

Reply:
I am also confused on how to use ?? here.
I want to check parameter.Name but I don't want to set that value. Instead I want to set the value to be parameter.Name.Name